src/test/java/com/billingapi/controller/InvoiceControllerTest.java

```diff
@@ -0,0 +1,586 @@
+               .willAnswer((invocation) -> invocation.getArgument(0));
+
+       // when - action or behaviour that we are going test
+       ResultActions response = mockMvc.perform(post("/invoice")
+               .contentType(MediaType.APPLICATION_JSON)
+               .content(objectMapper.writeValueAsString(newInvoice)));
+
+       // then - verify the result or output using assert statements
+       response.andDo(print()).
+               andExpect(status().isBadRequest());
+
+
+   }
+    /* Negative scenario Test to Create invoice if Vat Is Greater Than 99 then it throws exception  MethodArgumentNotValidException
+     */
+   @Test
+   public void should_throw_exception_whenCreateInvoice_ifvatIsGreaterThan99_thenReturnaddInvoiceFailed() throws Exception {
+       InvoiceRequest newInvoice = InvoiceRequest.builder()
+               .itemName("jeans")
+               .price(new BigDecimal("98000.0"))
+               .tax(10.00)
+               .vat(150.00)
+               .discount(20.00)
+               .isItemOnSale(true)
+               .discountOnSale(10.00)
+               .quantity(5)
+               .description("good")
+               .build();
+
+       given(invoiceService.save(any(Invoice.class)))
+               .willAnswer((invocation) -> invocation.getArgument(0));
+
+       // when - action or behaviour that we are going test
+       ResultActions response = mockMvc.perform(post("/invoice")
+               .contentType(MediaType.APPLICATION_JSON)
+               .content(objectMapper.writeValueAsString(newInvoice)));
+
+       // then - verify the result or output using assert statements
+       response.andDo(print()).
+               andExpect(status().isBadRequest());
+
+
+   }
+
+
+    /* Negative scenario Test to Create invoice if Discount  Is Greater Than 99 then it throws exception  MethodArgumentNotValidException
+     */
+   @Test
+   public void should_throw_exception_whenCreateInvoice_ifDiscountIsGreaterThan99_thenReturnaddInvoiceFailed() throws Exception {
+       InvoiceRequest newInvoice = InvoiceRequest.builder()
+               .itemName("jeans")
+               .price(new BigDecimal("98000.0"))
+               .tax(10.00)
+               .vat(15.00)
+               .discount(100.00)
+               .isItemOnSale(true)
+               .discountOnSale(10.00)
+               .quantity(5)
+               .description("good")
+               .build();
+
+       given(invoiceService.save(any(Invoice.class)))
+               .willAnswer((invocation) -> invocation.getArgument(0));
+       // when - action or behaviour that we are going test
+       ResultActions response = mockMvc.perform(post("/invoice")
+               .contentType(MediaType.APPLICATION_JSON)
+               .content(objectMapper.writeValueAsString(newInvoice)));
+       // then - verify the result or output using assert statements
+       response.andDo(print()).
+               andExpect(status().isBadRequest());
+
+   }
+
```

Comment:
You can use parameterized test & MethodSource annotation , because similar test are repetative.